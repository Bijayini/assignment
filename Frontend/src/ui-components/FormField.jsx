import React from 'react';
import PropTypes from 'prop-types';

const FormField = ({label, labelFor,children}) =>(<div className="form-field">
    <label htmlFor={labelFor}>{label} &#42;</label>
    {children}
</div>);

FormField.propTypes = {
  label: PropTypes.string.isRequired,
  labelFor: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default FormField;