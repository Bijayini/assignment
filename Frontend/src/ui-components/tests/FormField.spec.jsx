import React from 'react';
import { shallow } from 'enzyme';

import FormField from '../FormField';

describe('<FormField />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      label:'some label',
      labelFor:'someId',
      children:<span>Some input</span>
    };
    component = shallow(<FormField {...props} />);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});