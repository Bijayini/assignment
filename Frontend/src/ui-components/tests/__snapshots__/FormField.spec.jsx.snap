// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField /> should match with snapshot 1`] = `
"<div className=\\"form-field\\">
  <label htmlFor=\\"someId\\">
    some label
     *
  </label>
  <span>
    Some input
  </span>
</div>"
`;
